libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -152,6 +156,8 @@ bool StaticAnalyzer::visit(MemberAccess const& _memberAccess)
 {
 	bool const v050 = m_currentContract->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050);
 
+	m_memberAccess = true;
```

Comment:
While it is nice that we can catch `(this).f()` using this mechanism, I think it would also generate an error on `f(this).f()`, which should in general not be the case. Constructors cannot have nested non-constructor functions, but a member access expression can, so I don't think this can be solved by using stateful AST traversal.

Reply:
This was exactly the problem I was just thinking about. However, "f(this).f()" does not seem to trigger the warning - I just added a test for that.

Reply:
Just trying to think of a case in which this will in fact fail.
Otherwise I could just check for ``this`` in the member access again - either while also checking for ``(this)`` or while ignoring cases like ``(this)``...

Reply:
Ah, ok, now I got one that fails - I'll revert back to checking in member access.